### PR TITLE
Limits

### PR DESCRIPTION
**Pull request by Rohit-PX**
Why do we need this PR?

- To recreate a customer issue with the following config.

1. 15 node KOPs cluster with 5 nodes in each region.
2. 1 storage node and 4 storageless node per region 
3.  10 shared and 5 shared volumes per app on storage nodes
4. Create PX disk pool of 2 TB per node
5. Use scale factor of 10 to run tests

### CODE REVIEW

File: drivers/scheduler/k8s/specs/mysql-simplified/px-mysql-app.yaml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+          periodSeconds: 10
+          timeoutSeconds: 5
+        volumeMounts:
+        - name: mysql-persistent-storage-1
+          mountPath: /var/lib/mysql
+      volumes:
+      - name: mysql-persistent-storage-1
+        persistentVolumeClaim:
+          claimName: mysql-data-1
+      - name: mysql-persistent-storage-2
```
Comment (harsh-px) [inline drivers/scheduler/k8s/specs/mysql-simplified/px-mysql-app.yaml]: You don't need the rest of the volumes here. Just the first one.

File: drivers/scheduler/k8s/specs/nginx-shared-v4-simplified/px-nginx-app.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+kind: Service
+apiVersion: v1
+metadata:
+  name: nginx-service
+spec:
+  selector:
+    app: nginx
```
Comment (harsh-px) [inline drivers/scheduler/k8s/specs/nginx-shared-v4-simplified/px-nginx-app.yaml]: Change NodePort to ClusterIP

File: drivers/scheduler/k8s/specs/nginx-simplified/px-nginx-app.yaml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        volumeMounts:
+        - name: nginx-persistent-storage
+          mountPath: /usr/share/nginx/html
+      volumes:
+      - name: nginx-persistent-storage
+        persistentVolumeClaim:
+          claimName: px-nginx-pvc-1
+      - name: nginx-persistent-storage
+        persistentVolumeClaim:
+          claimName: px-nginx-pvc-2
```
Comment (harsh-px) [inline drivers/scheduler/k8s/specs/nginx-simplified/px-nginx-app.yaml]: Same as above. Only the first one is needed

File: drivers/scheduler/k8s/specs/nginx-shared-v4-simplified/px-nginx-app.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        volumeMounts:
+        - name: nginx-persistent-storage
+          mountPath: /usr/share/nginx/html
+        - name: nginx-persistent-storage-enc
+          mountPath: /usr/share/nginx/html-enc
+      volumes:
+      - name: nginx-persistent-storage
+        persistentVolumeClaim:
+          claimName: px-nginx-pvc
+      - name: nginx-persistent-storage-enc
```
Comment (harsh-px) [inline drivers/scheduler/k8s/specs/nginx-shared-v4-simplified/px-nginx-app.yaml]: enc PVC is not needed

File: drivers/scheduler/k8s/specs/nginx-shared-v4-simplified/px-nginx-app.yaml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        - containerPort: 80
+        volumeMounts:
+        - name: nginx-persistent-storage
+          mountPath: /usr/share/nginx/html
+        - name: nginx-persistent-storage-enc
+          mountPath: /usr/share/nginx/html-enc
+      volumes:
+      - name: nginx-persistent-storage
+        persistentVolumeClaim:
+          claimName: px-nginx-pvc
```
Comment (harsh-px) [inline drivers/scheduler/k8s/specs/nginx-shared-v4-simplified/px-nginx-app.yaml]: You are referencing the wrong PVC name here. Use the first PVC that you are creating.

Comment (Rohit-PX) [reply drivers/scheduler/k8s/specs/nginx-shared-v4-simplified/px-nginx-app.yaml]: Fixed.

File: drivers/scheduler/k8s/specs/nginx-shared-v4-simplified/px-nginx-app.yaml (hunk 4 above)
Comment (Rohit-PX) [reply drivers/scheduler/k8s/specs/nginx-shared-v4-simplified/px-nginx-app.yaml]: Fixed.

File: drivers/scheduler/k8s/specs/nginx-shared-v4-simplified/px-nginx-app.yaml (hunk 2 above)
Comment (Rohit-PX) [reply drivers/scheduler/k8s/specs/nginx-shared-v4-simplified/px-nginx-app.yaml]: Fixed.

File: drivers/scheduler/k8s/specs/mysql-simplified/px-mysql-app.yaml (hunk 1 above)
Comment (Rohit-PX) [reply drivers/scheduler/k8s/specs/mysql-simplified/px-mysql-app.yaml]: Fixed,

File: drivers/scheduler/k8s/specs/mysql-simplified/px-mysql-app.yaml [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+      maxSurge: 1
+      maxUnavailable: 1
+    type: RollingUpdate
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: mysql
+    spec:
+      schedulerName: stork
```
Comment (harsh-px) [inline drivers/scheduler/k8s/specs/mysql-simplified/px-mysql-app.yaml]: Remove `schedulerName: stork` from new specs as that will cause most pods to converge of the 3 storage nodes.

File: drivers/scheduler/k8s/specs/nginx-simplified/px-nginx-app.yaml [hunk 7]
```diff
@@ -0,0 +1,39 @@
+kind: Service
+apiVersion: v1
+metadata:
+  name: nginx-service
+spec:
+  selector:
+    app: nginx
+  type: NodePort
```
Comment (harsh-px) [inline drivers/scheduler/k8s/specs/nginx-simplified/px-nginx-app.yaml]: Change `type: NodePort` to `type: ClusterIP`

File: drivers/scheduler/k8s/specs/mysql-simplified/px-mysql-app.yaml (hunk 6 above)
Comment (Rohit-PX) [reply drivers/scheduler/k8s/specs/mysql-simplified/px-mysql-app.yaml]: Fixed.

File: drivers/scheduler/k8s/specs/nginx-simplified/px-nginx-app.yaml (hunk 7 above)
Comment (Rohit-PX) [reply drivers/scheduler/k8s/specs/nginx-simplified/px-nginx-app.yaml]: Fixed.

**Comment by harsh-px** [comment]
No longer needed. Was done to repro customer issue.

**Pull request closed without merging**